src/schema/v2/searchCriteriaLabel.ts

```diff
@@ -0,0 +1,346 @@
+import { GraphQLObjectType, GraphQLString } from "graphql"
+import { ResolverContext } from "types/graphql"
+import { startCase } from "lodash"
+
+import allAttributionClasses from "lib/attributionClasses"
+import { COLORS } from "lib/colors"
+
+type SearchCriteriaLabel = {
+  /** The GraphQL field name of the filter facet */
+  field: string
+
+  /** The human-friendly name of the filter facet */
+  name: string
+
+  /** The human-friendly value of the filter facet */
+  value: string
+}
+
+/**
+ * A type, derived here in Metaphysics from the upstream Gravity response,
+ * that represents a SearchCriteria's filter in a human-friendly label,
+ * suitable for use in a pill UI, for example.
+ */
+export const SearchCriteriaLabel = new GraphQLObjectType<
+  SearchCriteriaLabel,
+  ResolverContext
+>({
+  name: "SearchCriteriaLabel",
+  description:
+    "Human-friendly representation of a single SearchCriteria filter",
+  fields: {
+    field: {
+      type: GraphQLString,
+      description: "The GraphQL field name of the filter facet",
+    },
+    name: {
+      type: GraphQLString,
+      description: "The human-friendly name of the filter facet",
+    },
+    value: {
+      type: GraphQLString,
+      description: "The human-friendly value of the filter facet",
+    },
+  },
+})
+
+/**
+ * A resolver that takes the current SearchCriteria and returns a list of
+ * SearchCriteriaLabels representing that SearchCriteria in human-friendly form
+ */
+export const resolveSearchCriteriaLabels = async (
+  parent,
+  _args,
+  context,
+  _info
+) => {
+  const {
+    artistIDs,
+    attributionClass,
+    additionalGeneIDs,
+    priceRange,
+    sizes,
+    width,
+    height,
+    acquireable,
+    atAuction,
+    inquireableOnly,
+    offerable,
+    materialsTerms,
+    locationCities,
+    majorPeriods,
+    colors,
+    partnerIDs,
+  } = parent
+
+  const { artistLoader, partnerLoader } = context
+
+  const labels: any[] = []
```

Comment:
Just confirming my understanding: is it correct to say that we can't type this from the get-go because we're not sure what type `getArtistLabels` and `getPartnerLabels` will return?

Reply:
Good q — it sure seems like `labels` should be typed as `const labels: SearchCriteriaLabel[]`

But that ends up being incorrect because, as you suspect, some of those `get*` methods return a simple `{name, value, field}` object and some return an array of such objects. (Some even may return `undefined`)

It's only after that resulting `labels` array gets compacted and flattened on L100 that we can really assert that it is `SearchCriteriaLabel[]`

Prior to that it's technically `SearchCriteriaLabel | SearchCriteriaLabel[] | undefined)[]` which seemed ugly and not helpful — but idk maybe that is actually more informative than `any`?